src/server/game/AI/SmartScripts/SmartScript.cpp

```diff
@@ -3248,6 +3249,16 @@ void SmartScript::ProcessAction(SmartScriptHolder& e, Unit* unit, uint32 var0, u
 
             break;
         }
+        case SMART_ACTION_SET_WORLD_STATE:
+        {
+            sWorld->setWorldState(e.action.worldState.index, e.action.worldState.value);
```

Comment:
SmartScripts processed in the map thread, right? Then it's not unsafe to interact with world state map without synchronization mechanisms.

Reply:
this sets world state via `sWorld` instance, not `sWorldState`
I'm not sure if this requires a guard when called via SmartAI. This action can be omitted from this PR as it's not used. I left it unchanged

 link to `World::setWorldState()`
https://github.com/azerothcore/azerothcore-wotlk/blob/47c61b2576a5c86b405b6dce60b1daf4d69b93ca/src/server/game/World/World.cpp#L3150

Reply:
At the moment World::setWorldState interacts with simple std::map without locks or anything, so I don't think this code is safe.